waku-org/waku-interop-tests
# E2E_Tests_part1

## PR Details


the PR contain
1- 4 tests 
2- fix indentation issue at old test **test_time_filter_negative_start_time**


 ** 1- test_store_filter_interaction_with_six_nodes** does the following : 
 -  impelemntation of test mentioned in 
 - [example](<https://github.com/vacp2p/rfc-index/blob/main/waku/standards/core/10/waku2.md#overview-of-protocol-interaction>)

  1.   create 6 nodes with configurations according to what mentioned in the example
  2. start 5 nodes 
  3. subscribe 4 nodes to relay for topic "/waku/2/rs/3/1"  and 1 node for topic ""/waku/2/rs/3/9""
  4. node1 create and publish message
  5. node6 send filter request for the messages published on content topic ""/test/1/waku-filter/proto" "  pubsubtopic" "/waku/2/rs/3/1" " 
  6. check is done to validate the response of the filter request 
  7. node 5 goes alive 
  8. node5 makes store request to get published messages 
  9. check is done to validate response of store request 

**2- test_repeated_filter_requestID** 
  this test is done based on the following part of specs 
 
![Screenshot from 2024-10-29 16-09-54](https://github.com/user-attachments/assets/eabc0a5e-0292-48b8-8647-763198b6af99)


1-6 nodes created 
2- node1 publish message
3- node6 makes filter request 2 times with same request ID to node4
4- check is done to found the response 
5- at this point the expected response isn't clear [pending on devs reply]

**3-test_msg_stored_when_ephemeral_false**
**4-test_msg_not_stored_when_ephemeral_true**
  
the 2 tests are based on this part of specs 
![Screenshot from 2024-10-30 14-51-22](https://github.com/user-attachments/assets/dc1dd1a0-06b7-46ad-aa91-08d577d0d21d)

 1- 3 nodes created
  2- node 1 send message with ephemral flag = true 
   message is relayed to node2 
 3- check is done to verify that mesage isn't retrieved by store request done by node3

test_msg_stored_when_ephemeral_false  is the negative scenario when ephemeral flag = false 

message shall be stored 

  




## Issues reported:
1- Issue in repeated requested ID conflict in behavior when NODE_2 = GO-WAKU  repeated filter ID isn't allowed 
but when NODE_2 = NWAKU it works 

2- When NODE_2= GO-WAKU  and  flag ephemeral =True , error " **protocol not supported "** rises
